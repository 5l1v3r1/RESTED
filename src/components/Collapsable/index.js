import React, { PropTypes } from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { Button, Collapse, Row, Col } from 'react-bootstrap';

import Fonticon from 'components/Fonticon';
import * as Actions from 'store/config/actions';
import { isOpen } from 'store/config/selectors';

import { StyledCollapsable } from './StyledComponents';

export function Collapsable({ id, title, open, children, toggleCollapse }) {
  return (
    <StyledCollapsable>
      <Button
        bsStyle="link"
        onClick={e => {
          e.preventDefault();
          toggleCollapse(id, open);
        }}
      >
        <h4>
          {title}
          <Fonticon
            icon="angle-right"
            className={classNames({
              'fa-rotate-90': open,
            })}
          />
        </h4>
      </Button>
      <Collapse in={open}>
        <Row>
          <Col xs={12}>
            {children}
          </Col>
        </Row>
      </Collapse>
    </StyledCollapsable>
  );
}

Collapsable.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  toggleCollapse: PropTypes.func.isRequired,
  open: PropTypes.bool,
};

const mapStateToProps = (state, props) => ({
  open: isOpen(state, props),
});

export default connect(mapStateToProps, Actions)(Collapsable);

