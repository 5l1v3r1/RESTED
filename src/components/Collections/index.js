import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Panel } from 'react-bootstrap';

import * as Actions from 'store/collections/actions';
import CollectionList from './CollectionList';

function Titlebar({ addCollection }) {
  return (
    <span className="clearfix pull-right">
      <button onClick={addCollection}>
        Add new collection
      </button>
    </span>
  );
}

Titlebar.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

function Collections({ addCollection }) {
  return (
    <Panel header={<Titlebar addCollection={addCollection} />}>
      <CollectionList />
    </Panel>
  );
}

Collections.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

export default connect(null, Actions)(Collections);

