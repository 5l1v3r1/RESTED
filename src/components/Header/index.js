import React from 'react';
import { StyledHeader } from './StyledComponents';

export default function Header() {
  return (
    <StyledHeader>
      <h1>
        <img
          className="logo"
          role="presentation"
          height="40"
          src="img/rested-logo.png"
        />
        <span>RESTED</span>
      </h1>
    </StyledHeader>
  );
}

