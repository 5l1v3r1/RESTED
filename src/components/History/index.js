import React from 'react';

import { Panel } from './StyledComponents';
import HistoryList from './HistoryList';

export default function History() {
  return (
    <Panel header="">
      <HistoryList />
    </Panel>
  );
}

