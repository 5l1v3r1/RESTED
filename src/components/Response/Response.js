import React, { PropTypes } from 'react';
import { connect } from 'react-redux';

import * as Actions from 'store/request/actions';
import { isWrapResponse } from 'store/options/selectors';
import responsePropTypes, { responseShape } from 'propTypes/response';
import getContentType from 'utils/contentType';

import { StyledResponse, StyledHeader, Status } from './StyledComponents';
import Headers from './Headers';
import RenderedResponse from './RenderedResponse';
import FormattedResponse from './FormattedResponse';

function Titlebar({ url, time }) {
  return (
    <StyledHeader>
      <h3>
        Response ({time}) - <a href={url} className="text-muted">{url}</a>
      </h3>
    </StyledHeader>
  );
}

Titlebar.propTypes = {
  url: responseShape.url,
  time: PropTypes.node.isRequired,
};

export function Response(props) {
  const {
    response,
    wrapResponse,
    redirectChain,
    interceptedResponse,
  } = props;

  if (!response || !interceptedResponse) return null;

  const { method, url, totalTime, body } = response;

  const contentType = interceptedResponse.responseHeaders.find(header => (
    header.name.toLowerCase() === 'content-type'
  ));

  const type = getContentType(contentType && contentType.value);

  let time;
  if (redirectChain.length > 0) {
    time = `${(interceptedResponse.time / 1000).toFixed(3)}s, total time ${totalTime / 1000}s`;
  } else {
    time = `${totalTime / 1000}s`;
  }

  return (
    <StyledResponse
      wrapResponse={wrapResponse}
      header={<Titlebar method={method} url={url} time={time} />}
    >
      <h3>
        <Status
          green={response.status >= 200 && response.status < 300}
          red={response.status >= 400 && response.status < 600}
        >
          {response.status}
        </Status>
        <small> {response.statusText}</small>
      </h3>

      <Headers headers={interceptedResponse.responseHeaders} />
      {type.html && <RenderedResponse html={body} />}
      <FormattedResponse contentType={contentType} type={type} {...props} />
    </StyledResponse>
  );
}

Response.propTypes = {
  response: responsePropTypes,
  wrapResponse: PropTypes.bool.isRequired,
  redirectChain: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  interceptedResponse: PropTypes.shape({}),
};

const mapStateToProps = state => ({
  wrapResponse: isWrapResponse(state),
});

export default connect(mapStateToProps, Actions)(Response);

