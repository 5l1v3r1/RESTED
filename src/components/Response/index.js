import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Panel, Alert } from 'react-bootstrap';
import Highlight from 'react-highlight';

import * as Actions from 'store/request/actions';
import responsePropTypes, { responseShape } from 'propTypes/response';

import { StyledResponse, StyledHeader, Status } from './StyledComponents';
import Loading from './Loading';
import Headers from './Headers';

function Titlebar({ url, time }) {
  return (
    <StyledHeader>
      <h3>
        Response ({time / 1000}s) - <a href={url} className="text-muted">{url}</a>
      </h3>
    </StyledHeader>
  );
}

Titlebar.propTypes = {
  url: responseShape.url,
  time: responseShape.time,
};

export function Response({ response, loading }) {
  if (loading) {
    return (
      <Panel>
        <Loading />
      </Panel>
    );
  }

  if (!response) return null;

  const { method, url, headers, body, time } = response;

  // Some responses don't include the Content-Length header, in those cases
  // we simply allow it to be highligthed, for better or worse
  const contentLength = headers.find(header => (
    header.name.toLowerCase() === 'content-length'
  ));

  return (
    <StyledResponse header={<Titlebar method={method} url={url} time={time} />}>
      <h3>
        <Status
          green={response.status >= 200 && response.status < 300}
          red={response.status >= 400 && response.status < 600}
        >
          {response.status}
        </Status>
        <small> {response.statusText}</small>
      </h3>
      <Headers headers={headers} />
      {!contentLength || Number(contentLength.value) < 20000
        ? (
          <Highlight>
            {body}
          </Highlight>
        ) : (
          <span>
            <Alert bsStyle="warning">
              The size of the response is greater than 20KB, syntax
              highlighting has been disabled for performance reasons.
            </Alert>

            <code><pre>
              { body }
            </pre></code>
          </span>
        )
      }
    </StyledResponse>
  );
}

Response.propTypes = {
  loading: PropTypes.bool.isRequired,
  response: responsePropTypes,
};

const mapStateToProps = ({ request: { loading, response } }) => ({
  response,
  loading,
});

export default connect(mapStateToProps, Actions)(Response);

