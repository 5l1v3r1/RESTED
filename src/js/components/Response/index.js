import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Panel } from 'react-bootstrap';
import Highlight from 'react-highlight';

import Loading from './Loading';
import Headers from './Headers';
import * as Actions from '../../store/request/actions';
import responsePropTypes, { responseShape } from '../../propTypes/response';

function Titlebar({ url }) {
  return (
    <h3>
      Response - <a href={url}>{url}</a>
    </h3>
  );
}

Titlebar.propTypes = {
  url: responseShape.url,
};

export function Response({ response, loading }) {
  console.log('response', response);
  if (loading) {
    return (
      <Panel>
        <Loading />
      </Panel>
    );
  }

  if (!response) return null;

  const { method, url, headers, body } = response;
  return (
    <Panel header={<Titlebar method={method} url={url} />}>
      <h3>
        {response.status}
        <small>{response.statusText}</small>
      </h3>
      <Headers headers={headers} />
      <Highlight>
        {body}
      </Highlight>
    </Panel>
  );
}

Response.propTypes = {
  loading: PropTypes.bool.isRequired,
  response: responsePropTypes,
};

const mapStateToProps = ({ request: { loading, response } }) => ({
  response,
  loading,
});

export default connect(mapStateToProps, Actions)(Response);

