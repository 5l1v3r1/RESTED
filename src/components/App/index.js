import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Row } from 'react-bootstrap';
import HTML5Backend from 'react-dnd-html5-backend';
import { DragDropContext } from 'react-dnd';
import flow from 'lodash.flow';

import Header from 'components/Header';
import LeftPanel from 'components/LeftPanel';
import Request from 'components/Request';
import Response from 'components/Response';
import Modal from 'components/Modal';
import updateTheme from 'utils/updateTheme';
import { getTheme, getCollectionsMinimized } from 'store/options/selectors';
import { fetchOptions } from 'store/options/actions';
import { fetchUrlVariables } from 'store/urlVariables/actions';
import { THEMES } from 'constants/constants';
import './GlobalStyles';

import { LeftCol, RightCol } from './StyledComponents';

/*
 * This must be a React.Component because DragDropContext
 * attaches a ref to the component, which as we know will
 * not work with a stateless functional component.
 */
class App extends React.Component {
  constructor(props) {
    super(props);
    props.fetchOptions();
    props.fetchUrlVariables();

    updateTheme(props.theme);
  }

  componentWillReceiveProps(newProps) {
    if (newProps.theme !== this.props.theme) {
      updateTheme(newProps.theme);
    }
  }

  render() {
    const { collectionsMinimized } = this.props;

    return (
      <div>
        <Header />
        <Row>
          <LeftCol
            xsHidden
            sm={collectionsMinimized ? null : 4}
            collapsed={collectionsMinimized}
          >
            <aside>
              <LeftPanel />
            </aside>
          </LeftCol>
          <RightCol
            xs={12}
            sm={collectionsMinimized ? 12 : 8}
          >
            <main>
              <Request />
              <Response />
            </main>
          </RightCol>
        </Row>
        <Modal />
      </div>
    );
  }
}

App.propTypes = {
  fetchUrlVariables: PropTypes.func.isRequired,
  fetchOptions: PropTypes.func.isRequired,
  theme: PropTypes.oneOf(THEMES).isRequired,
  collectionsMinimized: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  theme: getTheme(state),
  collectionsMinimized: getCollectionsMinimized(state),
});

export default flow(
  connect(mapStateToProps, { fetchOptions, fetchUrlVariables }),
  DragDropContext(HTML5Backend),
)(App);

