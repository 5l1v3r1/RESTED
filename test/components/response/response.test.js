import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

/* eslint-disable import/no-unresolved */
import { Response } from 'components/Response';
import makeStore from '../../makeStore';

jest.mock('react-highlight');

describe('response component', () => {
  const response = {
    url: 'http://example.com',
    method: 'GET',
    status: 200,
    statusText: 'OK',
    body: 'Some long body',
    headers: [{
      name: 'Content-Type',
      value: 'application/awesome',
    }],
  };

  const store = makeStore({
    request: { response },
  });

  it('renders nothing given no props', () => {
    const tree = renderer.create(
      <Response loading={false} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders a loading gif when requets is in flight', () => {
    const tree = mount(
      <Provider store={store}>
        <Response response={response} loading />
      </Provider>,
    );

    expect(!tree.find('Loading').isEmpty()).toBe(true);
  });

  it('renders a result when given one', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Response response={response} loading={false} />
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('displays the status and statusText', () => {
    const tree = mount(
      <Provider store={store}>
        <Response response={response} loading={false} />
      </Provider>,
    );

    const h3 = tree.find('.panel-body h3');
    expect(h3.prop('children')[0]).toEqual(200);

    const small = h3.find('small');
    expect(small.prop('children')).toEqual('OK');
  });

  it('displays the URL and method in the titlebar', () => {
    const tree = mount(
      <Provider store={store}>
        <Response response={response} loading={false} />
      </Provider>,
    );

    const expectedLink = (
      <a href="http://example.com">
        http://example.com
      </a>
    );

    const heading = tree.find('.panel-heading');
    expect(heading.contains(expectedLink)).toEqual(true);
  });
});

