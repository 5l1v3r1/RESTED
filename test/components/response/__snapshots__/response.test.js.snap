// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`response component renders a result when given one 1`] = `
<div
  className="panel panel-default"
  id={undefined}
>
  <div
    className="panel-heading"
  >
    <h3>
      Response - 
      <a
        href="http://example.com"
      >
        http://example.com
      </a>
    </h3>
  </div>
  <div
    className="panel-body"
  >
    <h3>
      200
      <small>
        OK
      </small>
    </h3>
    <div
      className="gbeLmT"
    >
      <button
        className="btn btn-link"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <h4>
          Headers
          <i
            className="fa fa-angle-right"
            role="presentation"
          />
        </h4>
      </button>
      <div
        aria-expanded={null}
        className="collapse col-xs-12"
      />
    </div>
  </div>
</div>
`;

exports[`response component renders nothing given no props 1`] = `null`;
